src/App.tsx

```diff
@@ -266,6 +312,27 @@ function App() {
         alert(`An error occurred ${e || "unknown"}`);
       });
   }, [keyFobId, getFobUserStats]);
+
+  const onAddressScanned = useCallback(async (address: string) => {
+    setQrCodeData(address);
+
+    // If this is Hype transaction, resolve HL Name (Primary Name)
+    if (nextPageState?.data?.chain?.name === "Hype") {
+      try {
+        const statsData = await fetch(`${ATM_BACKEND_URL}/stats`).then(x => x.json());
+        if (statsData.resolveHypeName) {
+          const hlName = await statsData.resolveHypeName(address);
+          setRecipientHLName(hlName);
+        }
+      } catch (error) {
+        console.error("Error resolving HL name:", error);
+      }
+    }
+
+    setNextPageState({ state: PageState.BUYING_HYPE_INSERT_BILL, data: nextPageState.data });
+    setIsConfirmAddressDialogOpen(true);
```

Comment:
remove all the future state prep either from here or from where onAddressScanned is called from